documentation/asciidoc/computers/raspberry-pi/boot-eeprom-rpi4.adoc

```diff
@@ -183,29 +183,43 @@ rpi-eeprom-config --config boot.conf --out new.bin pieeprom.bin
+At power on, the BCM2711 ROM looks for a file called `recovery.bin` in the root directory of the boot partition on the SD card. If a valid `recovery.bin` is found then the ROM executes this instead of the contents of the EEPROM. This mechanism ensures that the bootloader EEPROM can always be reset to a valid image with factory default settings.
 
 See also xref:raspberry-pi.adoc#raspberry-pi-4-boot-flow[Raspberry Pi 4 boot-flow]
 
 ==== EEPROM update files
 
-| Filename | Purpose |
-|-------|------|
-| recovery.bin | VideoCore EEPROM recovery executable |
-| pieeprom.upd | Bootloader EEPROM image |
-| pieeprom.bin | Bootloader EEPROM image - same as pieeprom.upd but changes recovery.bin behaviour |
-| pieeprom.sig | The sha256 checksum of bootloader image (pieeprom.upd/pieeprom.bin) |
-| vl805.bin | The VLI805 USB firmware EEPROM image - ignored on 1.4 board revision which does not have a dedicated VLI EEPROM|
-| vl805.sig | The sha256 checksum of vl805.bin |
-
-* If the bootloader update image is called `pieeprom.upd` then `recovery.bin` renames itself to `recovery.000` and resets the CPU. Since `recovery.bin` is no longer present the ROM loads the newly updated bootloader from SPI EEPROM and the OS is booted as normal.
-* If the bootloader update image is called `pieeprom.bin` the `recovery.bin` will stop after the update has completed. On success the HDMI output will be green and the green activity LED is flashed rapidly. Otherwise, the HDMI output will be red and an xref:configuration.adoc#led-warning-flash-codes[error code] will be displayed via the activity LED.
-* The `.sig` files should just contain the sha256 checksum (in hex) of the corresponding image file. Other fields may be added in the future.
-* The BCM2711 ROM does not support loading `recovery.bin` from USB mass storage or TFTP. Instead, newer versions of the bootloader support a self-update mechanism where the SPI bootloader is able to reflash the SPI EEPROM itself. See `ENABLE_SELF_UPDATE` on the xref:raspberry-pi.adoc#raspberry-pi-4-bootloader-configuration[bootloader configuration] page.
+[cols="1,1"]
+|===
+| Filename
+| Purpose
+
+| recovery.bin
+| bootloader EEPROM recovery executable
+
+| pieeprom.upd
+| Bootloader EEPROM image
+
+| pieeprom.bin
+| Bootloader EEPROM image - same as pieeprom.upd but changes recovery.bin behaviour
+
+| pieeprom.sig
+| The sha256 checksum of bootloader image (pieeprom.upd/pieeprom.bin)
+
+| vl805.bin
+| The VLI805 USB firmware EEPROM image - ignored on 1.4 and later board revisions which do not have a dedicated VLI EEPROM
+
+| vl805.sig| The sha256 checksum of vl805.bin
+|===
+
+* If the bootloader update image is called `pieeprom.upd` then `recovery.bin` renames itself to `recovery.000` and resets the CPU. Since `recovery.bin` is no longer present the ROM loads the newly updated bootloader from EEPROM and the OS is booted as normal.
```

Comment:
"renames itself to `recovery.000`" -> "renames itself to `recovery.000` after the update has completed" ?

Reply:
I've reworded this. You may wish to have timg236 check this over to ensure the exact meaning of my various changes matches how things actually work.